plugins/catalog-backend/src/service/DefaultEntitiesCatalog.ts

```diff
@@ -256,9 +262,35 @@ export class DefaultEntitiesCatalog implements EntitiesCatalog {
           .select('refresh_state_references.source_entity_ref');
       });
 
+    // Stitch the entities that the deleted one had relations to. If we do not
+    // do this, the entities in the other end of the relations will still look
+    // like they have a relation to the entity that was deleted, despite not
+    // having any corresponding rows in the relations table.
+    const relationPeers = await this.database
+      .from<DbRelationsRow>('relations')
+      .innerJoin<DbRefreshStateReferencesRow>('refresh_state', {
+        'refresh_state.entity_ref': 'relations.target_entity_ref',
+      })
+      .where('relations.originating_entity_id', '=', uid)
+      .andWhere('refresh_state.entity_id', '!=', uid)
+      .select({ ref: 'relations.target_entity_ref' })
+      .union(other =>
+        other
+          .from<DbRelationsRow>('relations')
+          .innerJoin<DbRefreshStateReferencesRow>('refresh_state', {
+            'refresh_state.entity_ref': 'relations.source_entity_ref',
+          })
+          .where('relations.originating_entity_id', '=', uid)
+          .andWhere('refresh_state.entity_id', '!=', uid)
+          .select({ ref: 'relations.source_entity_ref' }),
+      );
+
+    // Perform the actual deletion
     await this.database<DbRefreshStateRow>('refresh_state')
       .where('entity_id', uid)
       .delete();
+
+    await this.stitcher.stitch(new Set(relationPeers.map(p => p.ref)));
```

Comment:
We need to do this for `replaceUnprocessedEntities` as well right?

Reply:
Maybe - will have to look into that separately in that case, gotta jump on another thing first

Reply:
A TODO? :grin: